File: backend/server/model/subscription.go
package model

import "time"

type SubscriptionPlan struct {
	ID            string `gorm:"type:uuid;primary_key;default:uuid_generate_v4()" json:"id"`
	Name          string `json:"name"`
	Description   string `gorm:"type:text" json:"description"`
	Duration      int    `json:"duration"`
	StripePriceID string `json:"stripe_price_id"`
}

type UserSubscription struct {
	ID                   string    `gorm:"type:uuid;primary_key;default:uuid_generate_v4()" json:"id"`
	UserID               string    `gorm:"type:uuid" json:"user_id"`
	PlanID               string    `gorm:"type:uuid" json:"subscription_id"`
	StartAt              time.Time `json:"start_at"`
	EndAt                time.Time `json:"end_at"`
	IsActive             bool      `json:"is_active"`
	StripeSubscriptionID string    `json:"stripe_subscription_id"`

	User *User             `json:"user"`
	Plan *SubscriptionPlan `json:"plan"`
}

File: frontend/ReSell/app/src/main/java/com/example/resell/model/Subscription.kt
package com.example.resell.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import java.time.OffsetDateTime

@JsonClass(generateAdapter = true)
data class SubscriptionPlan(
    @Json(name = "id")
    val id: String,

    @Json(name = "name")
    val name: String,

    @Json(name = "description")
    val description: String,

    @Json(name = "duration")
    val duration: Int,

    @Json(name = "stripe_price_id")
    val stripePriceId: String
)

@JsonClass(generateAdapter = true)
data class UserSubscription(
    @Json(name = "id")
    val id: String,

    @Json(name = "user_id")
    val userId: String,

    @Json(name = "subscription_id")
    val planId: String,

    @Json(name = "start_at")
    val startAt: OffsetDateTime,

    @Json(name = "end_at")
    val endAt: OffsetDateTime,

    @Json(name = "is_active")
    val isActive: Boolean,

    @Json(name = "stripe_subscription_id")
    val stripeSubscriptionId: String,

    @Json(name = "user")
    val user: User? = null,

    @Json(name = "plan")
    val plan: SubscriptionPlan? = null
)
